src/connectors/numeric-menu/connectNumericMenu.ts

```diff
@@ -154,10 +165,10 @@ export default function connectNumericMenu(renderFn, unmountFn = noop) {
       render({ results, state, instantSearchInstance }) {
         renderFn(
           {
-            createURL: this._createURL(state),
-            items: transformItems(this._prepareItems(state)),
+            createURL: connectorState.createURL!(state),
+            items: transformItems(connectorState.prepareItems!(state)),
             hasNoResults: results.nbHits === 0,
-            refine: this._refine,
+            refine: connectorState.refine!,
```

Comment:
why not use getRefine() here ?

Reply:
We need ! here because the object is initialized empty, and thus all keys are optional, some better options are outlined in #4102